src/app/qgisapp.cpp

```diff
@@ -2081,10 +2081,12 @@ void QgisApp::createToolBars()
 
   mSnappingWidget = new QgsSnappingWidget( QgsProject::instance(), mMapCanvas, mSnappingToolBar );
   mSnappingWidget->setObjectName( QStringLiteral( "mSnappingWidget" ) );
-  //mSnappingWidget->setFont( myFont );
   connect( mSnappingWidget, SIGNAL( snappingConfigChanged() ), QgsProject::instance(), SIGNAL( snappingConfigChanged() ) );
   mSnappingToolBar->addWidget( mSnappingWidget );
 
+  mTracer = new QgsMapCanvasTracer( mMapCanvas, messageBar() );
+  mTracer->setActionEnableTracing( mSnappingWidget->findChild<QAction*>( "EnableTracingAction" ) );
```

Comment:
This is a bit fragile - would be better to have api call e.g. mSnappingWidget->actionEnableTracing()